conans/test/integration/command_v2/list_package_ids_test.py

```diff
@@ -0,0 +1,180 @@
+import re
+
+import pytest
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient, TestServer
+
+
+class TestListPackageIdsBase:
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        self.client = TestClient()
+
+    def _add_remote(self, remote_name):
+        self.client.servers[remote_name] = TestServer(users={"username": "passwd"}, write_permissions=[("*/*@*/*", "*")])
+        self.client.update_servers()
+        self.client.run("user username -p passwd -r {}".format(remote_name))
+
+    def _create_recipe(self, reference):
+        self.client.save({'conanfile.py': GenConanfile()})
+        self.client.run("create . {}".format(reference))
+
+    def _upload_recipe(self, remote, reference):
+        self.client.save({'conanfile.py': GenConanfile()})
+        self.client.run("export . {}".format(reference))
+        self.client.run("upload --force -r {} {}".format(remote, reference))
+
+
+class TestParams(TestListPackageIdsBase):
+    def test_fail_if_reference_is_not_correct(self):
```

Comment:
We are not testing a reference that does not exist in cache but has a valid format, like:
`conan list package-ids "notexisting/1.0@user/channel#whatever_not_existing"`
This is not returning an error.

Reply:
@czoido we're doing that plenty of times along the test suite. For example, `test_by_default_search_only_in_cache` is returning:

```
        Local Cache:
          There are no matching recipes
```
So it should be a valid test 😉  This test is done for cache and remote as well.

Reply:
That test is failing in the PR: https://ci.conan.io/blue/organizations/jenkins/ConanTestSuitev2/detail/PR-9310/2/pipeline/34
Anyway, I think if you want to check the package-id's of a specific recipe revision that does not exist in the cache, we should raise an error not returning that there are not matching recipes.

Reply:
I tend to say that this check could go in another PR because it should be something that affects to all the list context, not only the `package-ids` subcommand